bluepysnap/circuit_ids.py

```diff
@@ -165,105 +176,166 @@ def get_ids(self, unique=False):
+        return self._apply(lambda x: x[sampling], inplace)
 
     def limit(self, limit_size, inplace=False):
-        """Limit the size of a CircuitNodeIds.
+        """Limit the size of a CircuitIds.
 
         Notes:
             this function takes the first ``limit_size`` values of a circuit node ids.
 
         Args:
             limit_size (int): the size of the sample. If the size of the sample is greater than
-                the size of the CircuitNodeIds then all ids are kept.
+                the size of the CircuitIds then all ids are kept.
             inplace (bool): if set to True. Do the transformation inplace.
         """
-        return self._slice_index(slice(0, limit_size), inplace=inplace)
+        return self._apply(lambda x: x[:limit_size], inplace)
+
+    def unique(self, inplace=False):
+        """Returns only unique CircuitIds.
+
+        Notes:
+            this function does not sort the ids
+        """
+        return self._apply(lambda x: x.unique(), inplace)
 
     def to_csv(self, filepath):
-        """Save CircuitNodeIds to csv format."""
+        """Save CircuitIds to csv format."""
         self.index.to_frame(index=False).to_csv(filepath, index=False)
 
     @classmethod
     def from_csv(cls, filepath):
-        """Load CircuitNodeIds from csv."""
+        """Load CircuitIds from csv."""
         return cls(pd.MultiIndex.from_frame(pd.read_csv(filepath)))
 
     def tolist(self):
-        """Convert the CircuitNodeIds to a list of tuples."""
+        """Convert the CircuitIds to a list of tuples."""
         return self.index.tolist()
 
+    def __call__(self, *args, **kwargs):
+        """Allows to use the CircuitIds as normal indices in a DataFrame."""
+        return self.index
+
+    def __len__(self):
+        """Return the length of the CircuitIds."""
+        return len(self.index)
+
     def __repr__(self):
-        """Correct repr of the CircuitNodeIds."""
-        return repr(self.index).replace("MultiIndex", "CircuitNodeIds", 1)
+        """Correct repr of the CircuitIds."""
+        return repr(self.index).replace("MultiIndex", self.__class__.__name__, 1)
 
     def __eq__(self, other):
-        """Equality for the CircuitNodeIds."""
+        """Equality for the CircuitIds."""
         if self is other:
             return True
-        if not isinstance(other, CircuitNodeIds):
+        if not isinstance(other, type(self)):
             return False
         return self.index.equals(other.index)
 
-    def __len__(self):
-        """Return the length of the CircuitNodeIds."""
-        return len(self.index)
+    def __iter__(self):  # pragma: no cover
+        """Iterator on the CircuitIds."""
+        raise NotImplementedError
+
+    def __getitem__(self, item):  # pragma: no cover
+        """Getter on the CircuitIds."""
+        raise NotImplementedError
+
+
+class CircuitNodeIds(CircuitIds, metaclass=DocSubstitutionMeta,
```

Comment:
I chose the meta class over the decorator due to problems with types when using the decorator. 
When you `type(CircuitNodeIds)` with the decorator, the shown type is the type of the class wrapper ie: 
```
type(CircuitNodeIds)
<class 'bluepysnap._doctools.DocSubstitutionDecorator.__call__.<locals>.Wrapped'>
```
Which is confusing. So I finally chose to use a metaclass to do the factorization.